test/middleware/service.availability.middleware.spec.ts

```diff
@@ -109,4 +121,63 @@ describe("service availability middleware tests", () => {
 
     expect(response.text).not.toContain("Service offline - Apply to remove an overseas entity from the register");
   });
+
+  describe("Remove feature flag tests for remove specific pages", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    test(`Redirects to the next page whenever the remove flag is true`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_CANNOT_USE_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(302);
+    });
+
+    test(`Does not reach the ${REMOVE_CANNOT_USE_PAGE} when the remove flag is false`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(false); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_CANNOT_USE_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).not.toContain("You cannot apply to remove this overseas entity");
+      expect(resp.text).not.toContain("Service unavailable");
```

Comment:
we should also check that the correct page is displayed, so we would need a .toContain("Sorry, the service is unavailable")

Reply:
same comment applies to the rest of the 'Does not reach' tests

Reply:
Added a further check.

Reply:
Not sure what happed there must have backed them out. They should be present now.